common/schema/DHQauthor-TEI.rng

```diff
@@ -7460,6 +7460,14 @@ Elements 7.2.5. Speech Contents]</a:documentation>
           <ref name="dhq_idno"/>
         </choice>
       </zeroOrMore>
+      <pattern xmlns="http://purl.oclc.org/dsdl/schematron" xmlns:rng="http://relaxng.org/ns/structure/1.0" id="dhq-idno-article-id_matches_fn-constraint-rule-16">
+        <sch:rule xmlns="http://www.tei-c.org/ns/1.0" xmlns:mml="http://www.w3.org/1998/Math/MathML" xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#" xmlns:sch="http://purl.oclc.org/dsdl/schematron" xmlns:xi="http://www.w3.org/2001/XInclude" context="tei:idno[ @type eq 'DHQarticle-id']">
+          <sch:let name="basename" value="replace( base-uri(.), '^.*/([^/.]+)\.xml$','$1')"/>
+          <sch:assert test="$basename eq normalize-space(.)">
```

Comment:
per @juliaflanders change msg to “… should match …”, in which case should use a `<report ... ne ...>` instead.